app/lib/npmInstall.js

```diff
@@ -11,7 +12,7 @@ const installStatus = require('./installStatus')
  *
  * On failure, returns a rejected promise, so `retry` runs it again.
  */
-const npmInstall = (name, packagePath) => {
+const npmInstall = cooldown((name, packagePath) => {
```

Comment:
A unit test that installs and updates in the same test could be a good end to end test. There are already tests to do both, just not in the same test.